components/formats-gpl/src/loci/formats/in/MetamorphReader.java

```diff
@@ -871,12 +874,13 @@ else if (firstSeriesChannels != null) {
 
           if ((int) wave[waveIndex] >= 1) {
             // link LightSource to Image
+            int laserIndex = i * getEffectiveSizeC() + c;
```

Comment:
You're right (<a href="http://static.javadoc.io/org.openmicroscopy/ome-xml/5.4.0/ome/xml/meta/MetadataStore.html#setDetectorSettingsBinning-ome.xml.model.enums.Binning-int-int-">setDetectorSettingsBinning</a> and <a href="http://static.javadoc.io/org.openmicroscopy/ome-xml/5.4.0/ome/xml/meta/MetadataStore.html#setDetectorSettingsReadOutRate-ome.units.quantity.Frequency-int-int-">setDetectorSettingsReadOutRate</a> for external references). No need to change these callss